plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryConnectorTest.java

```diff
@@ -1077,6 +1083,25 @@ public void testNativeQueryIncorrectSyntax()
                 .failure().hasMessageContaining("Failed to get destination table for query");
     }
 
+    @Test
+    public void testQueriesWithParentProjectId()
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = BigQueryQueryRunner.builder()
+                .setConnectorProperties(ImmutableMap.<String, String>builder()
+                        .put("bigquery.arrow-serialization.enabled", "false")
+                        .put("bigquery.job.label-name", "trino_query")
+                        .put("bigquery.job.label-format", "q_$QUERY_ID__t_$TRACE_TOKEN")
+                        .put("bigquery.project-id", testingProjectId)
+                        .put("bigquery.parent-project-id", testingParentProjectId)
+                        .buildOrThrow())
+                .build();
+
+        assertThat(queryRunner.execute(getSession(), "SELECT name FROM bigquery.tpch.nation WHERE nationkey = 0").getOnlyValue()).isEqualTo("ALGERIA");
+        assertThat(queryRunner.execute(getSession(), "SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT name FROM tpch.nation WHERE nationkey = 0'))").getOnlyValue()).isEqualTo("ALGERIA");
+        assertThat(queryRunner.execute(getSession(), format("SELECT * FROM TABLE(bigquery.system.query(query => 'SELECT name FROM %s.tpch.nation WHERE nationkey = 0'))", testingProjectId)).getOnlyValue()).isEqualTo("ALGERIA");
```

Comment:
Can we use `computeScalar` method?

Reply:
After introducing new test class I followed `assertQuery()` convention.